### PR TITLE
feat(api): Add helpful hint for missing trailing slashes

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Looks like this:

    Route not found, did you forget a trailing slash?

    try: /api/0/test/
                    ^

It will also respond with json if you have `Accepts: application/json`

    {"info": "Route not found, did you forget a trailing slash? try: /api/0/test/"}

### CODE REVIEW

File: src/sentry/api/endpoints/catchall.py [hunk 1]
```diff
@@ -12,4 +12,19 @@ class CatchallEndpoint(Endpoint):
 
     @csrf_exempt
     def dispatch(self, request: Request, *args, **kwargs) -> Response:
+        """
+        This endpoint handles routes that did not match
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/catchall.py:16]:
```suggestion
        This endpoint handles routes that did not match under `/api/0/`
```

Comment (evanpurkhiser) [reply src/sentry/api/endpoints/catchall.py]: good

File: tests/sentry/api/endpoints/test_catchall.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+@control_silo_test
+class CatchallTestCase(APITestCase):
+    def setUp(self):
+        super().setUp()
+
+    def test_simple(self):
+        response = self.client.get("/api/0/bad_url/")
+        assert_status_code(response, status.HTTP_404_NOT_FOUND)
+
```
Comment (JoshFerge) [inline tests/sentry/api/endpoints/test_catchall.py:20]: assert `response.content` empty here?

Comment (evanpurkhiser) [reply tests/sentry/api/endpoints/test_catchall.py:20]: sure

**Review by JoshFerge — APPROVED** [review]

**Pull request merged**